### PR TITLE
(GH-14) Cross Platform Task

### LINKED ISSUE #14: Add support for non-Windows agents
Right now only Windows agents are supported. We should add support for non-Windows agents as well.

### PR DESCRIPTION
**Pull request by louisfischer**
Fixes #14 

This pull request is to begin the process of creating a cross-platform agent for Azure DevOps that will run on Windows and Non-Windows Agent machines. Please review and let me know if this approach will work or what you would like to see changed. The original Powershell bootstrap is still there without any modifications.

Below is an image of the script running using a basic build.cake script

image
image
image

### CODE REVIEW

**Comment by louisfischer** [comment]
@gep13 @devlead will one of you let me know if anything needs to be updated or reworked?

File: package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "cake-task",
+  "version": "1.0.0",
```
Comment (patriksvensson) [inline package.json:3]: Do we need to update this version before publishing?

File: src/cake.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        console.log('Bootstrapping build script...');
+        const bootstrapExitCode = await tl.exec(cakeToolPath, `"${$script}" --bootstrap`);
+        if (bootstrapExitCode !== 0) {
+            throw new Error(`Failed to bootstrap the build script. Exit code: ${bootstrapExitCode}`);
+        }
+    }
+
+    // Start Cake
+    console.log('Executing build script...');
+    const exitCode = await tl.exec(cakeToolPath, `"${$script}" -target="${$target}" -verbosity="${$verbosity}" --paths_tools="${toolPath}" ${$arguments}`);
```
Comment (patriksvensson) [inline src/cake.ts]: Perhaps change `${toolPath}` to `${$toolPath}` so it's consistent?

File: package.json (hunk 1 above)
Comment (louisfischer) [reply package.json:3]: @patriksvensson modified the build.cake script to version package.json to be consistent with the other versioning locations

File: src/cake.ts (hunk 2 above)
Comment (louisfischer) [reply src/cake.ts]: @patriksvensson done.

**Comment by louisfischer** [comment]
@patriksvensson @gep13 @devlead Any other suggestions or feedback you may have for this?

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@louisfischer I made a few changes to this, but on the whole, this was a great PR, thank you very much for providing this!

**Pull request merged**